pkg/services/libraryelements/database.go

```diff
@@ -882,3 +888,43 @@ func contains(slice []string, element string) bool {
 	}
 	return false
 }
+
+func getFoldersWithMatchingTitles(c context.Context, l *LibraryElementService, signedInUser identity.Requester, query model.SearchLibraryElementsQuery) ([]string, error) {
+	var foldersWithMatchingTitles []string
+	if len(strings.TrimSpace(query.SearchString)) > 0 {
+		searchQuery := folder.SearchFoldersQuery{
+			OrgID:        signedInUser.GetOrgID(),
+			Title:        query.SearchString,
+			SignedInUser: signedInUser,
+		}
+
+		if l.features.IsEnabled(c, featuremgmt.FlagKubernetesClientDashboardsFolders) {
+			folderHits, err := l.folderService.SearchFolders(c, searchQuery)
+
+			if err != nil {
+				return nil, err
+			}
+			foldersWithMatchingTitles = make([]string, 0, len(folderHits))
+			for _, hit := range folderHits {
+				foldersWithMatchingTitles = append(foldersWithMatchingTitles, hit.UID)
+			}
+		} else {
```

Comment:
Instead of an else we could return above and put al the else code aligned to the left below.